test/Altinn.Platform.Events.Tests/TestingControllers/EventsControllerTests.cs

```diff
@@ -491,20 +492,87 @@ public async Task Post_AuthorizedRequest_EventIsRegistered()
                 Assert.Equal(HttpStatusCode.OK, response.StatusCode);
             }
 
-            private HttpClient GetTestClient(IEventsService eventsService = null, bool enableExternalEvents = false)
+            [Fact]
+            public async Task Post_PlatformAccessTokenIncluded_EventIsRegistered()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}/events";
+
+                Mock<IEventsService> eventMock = new();
+                eventMock.Setup(em => em.RegisterNew(It.IsAny<CloudEvent>()))
+                          .ReturnsAsync(Guid.NewGuid().ToString());
+
+                Mock<IAuthorization> authorizationMock = new();
+                authorizationMock.Setup(a => a.AuthorizePublishEvent(It.IsAny<CloudEvent>()))
+                            .ReturnsAsync(true);
+
+                HttpClient client = GetTestClient(eventMock.Object, authorizationMock.Object, true);
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetOrgToken("ttd"));
+
+                HttpRequestMessage httpRequestMessage = new(HttpMethod.Post, requestUri)
+                {
+                    Content = new StringContent(_validEvent.Serialize(), Encoding.UTF8, "application/cloudevents+json")
+                };
+                httpRequestMessage.Headers.Add("PlatformAccessToken", PrincipalUtil.GetAccessToken("ttd", "apps-test"));
+
+                // Act
+                HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            }
+
+            [Fact]
+            public async Task Post_UserAuthenticatedNotAuthorized_ReturnsForbidden()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}/events";
+
+                Mock<IEventsService> eventMock = new();
+                eventMock.Setup(em => em.RegisterNew(It.IsAny<CloudEvent>()))
+                          .ReturnsAsync(Guid.NewGuid().ToString());
+
+                Mock<IAuthorization> authorizationMock = new();
+                authorizationMock.Setup(a => a.AuthorizePublishEvent(It.IsAny<CloudEvent>()))
+                            .ReturnsAsync(false);
+
+                HttpClient client = GetTestClient(eventMock.Object, authorizationMock.Object, true);
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetOrgToken("ttd"));
```

Comment:
Should this have been a user token instead of an org token? Ref the test name.

Reply:
Renamed test.